model/task/db.go

```diff
@@ -292,11 +298,16 @@ var (
 	AddAnnotations = []bson.M{
 		{
 			"$facet": bson.M{
-				// We skip annotation lookup for non-failed tasks, because these can't have annotations
+				// We skip annotation lookup for non-failed tasks, because these can't have annotations,
+				// and lookup is not necessary for tasks that already have HasAnnotationsKey set
+				// TODO: DEVPROD-___ simplify this step to only check HasAnnotationsKey
 				"not_failed": []bson.M{
 					{
 						"$match": bson.M{
-							StatusKey: bson.M{"$nin": evergreen.TaskFailureStatuses},
+							"$or": []bson.M{
+								{HasAnnotationsKey: true},
```

Comment:
I'm confused by what this does? 
https://github.com/hadjri/evergreen/blob/c5495d494db06a2fb5a0422194dddd1df63ccb0f/model/task/db.go#L2639 avoids this check if the task was stage if the version was created after this change.

Reply:
Yeah this would allow old versions to run the faster version of the pipeline should they get annotations added to them moving forward, but this is a negligible benefit / probably not worth worrying about. Changing it back to how it was.